src/06-mocking-node-api/index.test.ts

```diff
@@ -29,24 +44,53 @@ describe('doStuffByInterval', () => {
   });
 
   test('should set interval with provided callback and timeout', () => {
-    // Write your test here
+    const mock = jest.spyOn(global, 'setInterval');
+    const mls = 10000;
+    doStuffByInterval(clb, mls);
+
+    expect(mock).toHaveBeenCalledTimes(1);
+    expect(mock).toHaveBeenCalledWith(clb, mls);
   });
 
   test('should call callback multiple times after multiple intervals', () => {
-    // Write your test here
+    const COUNT = 10;
+    const mls = 1000;
+    const mockClb = jest.fn(clb);
+    jest.spyOn(global, 'setInterval');
+    doStuffByInterval(mockClb, mls);
+
+    for (let i = 1; i <= COUNT; i++) {
+      jest.advanceTimersByTime(mls);
+      expect(mockClb).toBeCalledTimes(i);
+    }
   });
 });
 
 describe('readFileAsynchronously', () => {
   test('should call join with pathToFile', async () => {
-    // Write your test here
+    const PATH_TO_FILE = 'test.txt';
+    const mock = jest.spyOn(path, 'join');
```

Comment:
Assignment tell "Remember that the tests should not interact with the actual file system and should not rely on real-time!"

only 3 point for this test because you use real nodejs function

if you put
`console.log('nodejs "join" function work proof: ', join('111', '222'));`
in 
`readFileAsynchronously`

you will see

```
  console.log
    nodejs "join" function work proof:  111\222
```
it is a mistake

you should use
`joinSpy.getMockImplementation(() => ...)`
or better
`jest.mock('path');`